src/app/frontend/deploy/deployfromfile_controller.js

```diff
@@ -33,5 +40,55 @@ export default class DeployFromFileController {
      * @export {{name:string, content:string}}
      */
     this.file = {name: '', content: ''};
+
+    /** @private {!angular.$q} */
+    this.q_ = $q;
+
+    /** @private {!angular.$resource} */
+    this.resource_ = $resource;
+
+    /** @private {!angular.$log} */
+    this.log_ = $log;
+
+    /** @private {!ui.router.$state} */
+    this.state_ = $state;
+
+    /**
+     * TODO (cheld) Set correct type after fixing issue #159
+     * @private {!Object}
+     */
+    this.errorDialog_ = errorDialog;
+  }
+
+  /**
+   * Deploys the application based on the state of the controller.
+   *
+   * @export
+   * @return {angular.$q.Promise}
+   */
+  deploy() {
+    /** @type {!backendApi.AppDeploymentFromFileSpec} */
+    let deploymentSpec = {
+      name: this.file.name,
+      content: this.file.content,
+    };
+
+    let defer = this.q_.defer();
+
+    /** @type {!angular.Resource<!backendApi.AppDeploymentFromFileSpec>} */
+    let resource = this.resource_('/api/appdeploymentfromfile');
```

Comment:
Dont start api calls with leading `/`. With this the app would not work when non rooted at the domain.